src/components/RegistrationForm/RegistrationForm.tsx

```diff
@@ -132,28 +136,48 @@ export const RegistrationForm = () => {
       return;
     }
 
-    setIsSubmitted(true);
-
-    // eslint-disable-next-line no-console
-    console.log('Form submitted:', formData);
+    try {
+      const supabase = getSupabaseClient();
+
+      const { error } = await supabase.auth.signUp({
+        email: formData.email,
+        password: formData.password,
+        options: {
+          data: {
+            first_name: formData.firstName,
+            last_name: formData.lastName,
+          },
+        },
+      });
+
+      if (error) {
+        console.error('Sign up error:', error);
+        if (error.message.includes('User already registered')) {
+          setEmailErrorMessage(ERROR_MESSAGES.emailExists);
+
+          return;
+        }
+
+        throw error;
+      }
```

Comment:
Dont nest IFs, if you can avoid it. Always strive towards "early return".

```suggestion
         if (error?.message.includes('User already registered')) {
              setEmailErrorMessage(ERROR_MESSAGES.emailExists);

             return;
         }
   
        if (error) {
            throw error;
      }
```

Reply:
ALSO! Error handling could be all part of the catch block, couldn't it?